File: pdu/python/hako_srv_msgs/pdu_cdr_conv_SystemControlRequest.py
from .pdu_pytype_SystemControlRequest import SystemControlRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_SystemControlRequest(writer: CdrWriter, src: SystemControlRequest):
    writer.write_uint32(src.opcode)


def cdr_body_to_py_SystemControlRequest(reader: CdrReader, dst: SystemControlRequest):
    dst.opcode = reader.read_uint32()
    return dst


def py_to_cdr_SystemControlRequest(src: SystemControlRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_SystemControlRequest(writer, src)
    return writer.bytes()


def cdr_to_py_SystemControlRequest(cdr_payload) -> SystemControlRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = SystemControlRequest()
    return cdr_body_to_py_SystemControlRequest(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_SystemControlRequest.js
import { SystemControlRequest } from './pdu_jstype_SystemControlRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduSystemControlRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { SystemControlRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.opcode);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { SystemControlRequest } dst
     * @returns { SystemControlRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.opcode = reader.read_uint32();
        return dst;
    }

    /**
     * @param { SystemControlRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { SystemControlRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new SystemControlRequest());
    }
}
